helm/prometheus-rules/templates/alerting-rules/gcp.management-cluster.rules.yml

```diff
@@ -0,0 +1,61 @@
+{{- if eq .Values.managementCluster.provider.kind "gcp" }}
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  creationTimestamp: null
+  labels:
+      {{- include "labels.common" . | nindent 4 }}
+    cluster_type: "management_cluster"
+  name: gcp.management-cluster.rules
+  namespace: {{ .Values.namespace  }}
+spec:
+  groups:
+  - name: gcp
+    rules:
+    - alert: ManagementClusterPodPendingGCP
+      annotations:
+        description: '{{`Pod {{ $labels.namespace }}/{{ $labels.pod }} is stuck in Pending.`}}'
+        opsrecipe: pod-stuck-in-pending/
+      expr: kube_pod_status_phase{namespace="giantswarm|kube-system",pod=~"(capi-controller-manager.*|capg-controller-manager.*|capi-kubeadm-bootstrap-controller-manager.*|capi-kubeadm-control-plane-controller-manager.*|cilium.*|csi-gce-pd-node.*)",phase="Pending"} == 1
+      for: 25m
+      labels:
+        area: kaas
+        cancel_if_outside_working_hours: "true"
+        cancel_if_cluster_status_creating: "true"
+        cancel_if_cluster_status_deleting: "true"
+        cancel_if_cluster_status_updating: "true"
+        cancel_if_kube_state_metrics_down: "true"
+        severity: page
+        team: hydra
+        topic: managementcluster
+    - alert: ManagementClusterContainerIsRestartingTooFrequentlyGCP
+      annotations:
+        description: '{{`Container {{ $labels.container }} in pod {{ $labels.namespace }}/{{ $labels.pod }} is restarting too often.`}}'
+        opsrecipe: container-is-restarting-too-often/
+      expr: increase(kube_pod_container_status_restarts_total{container=~"capi-controller-manager.*|capg-controller-manager.*|capi-kubeadm-bootstrap-controller-manager.*|capi-kubeadm-control-plane-controller-manager.*|cilium.*|csi-gce-pd-node.*"}[1h]) > 6
```

Comment:
Do we have other rules that alert if common pods shared with other providers are pending, like organization-operator, rbac-operator, etc?

Taking a look at our https://github.com/giantswarm/gcp-app-collection/tree/master/flux-manifests and https://github.com/giantswarm/default-apps-gcp/blob/main/helm/default-apps-gcp/values.yaml#L11-L151, I'd add a few more, i.e `dns-operator-gcp`, `workload-identity-operator-gcp`, etc.

Reply:
> I'd add a few more, i.e dns-operator-gcp, workload-identity-operator-gcp

Yeah, I wanted to include these two.

> Do we have other rules that alert if common pods shared with other providers are pending, like organization-operator, rbac-operator, etc?

We have more generalized `WorkloadClusterAppFailed`, `WorkloadClusterAppFailed{team name}` and `ManagementClusterAppFailed{team name}` which check App CR status.

Reply:
(not directly related for this PR) In general, we should try to come up with SLO-style alerts for as many components as possible, but for that we first have to define what we want to observe in all those components (what metric we care about) and then to figure out what SLO we want for those.

Reply:
I included dns-operator-gcp and workload-identity-operator-gcp, but not other apps. Let's start with these and then add more (even now it's more than other providers I think), so we are not immediately swamped with alerts.